tool/requote.js

```diff
@@ -17,22 +17,35 @@ const { DatasetParser, DatasetStringifier } = require('../lib/dataset-parsers');
 const { maybeCreateReadStream, readAllLines } = require('./lib/argutils');
 const StreamUtils = require('../lib/stream-utils');
 
-function findSubstring(sequence, substring) {
+const ENTITY_MATCH_REGEX = /^([A-Z].*)_[0-9]+$/;
+
+
+function check_range(index, spansBySentencePos){
+    for (let i = 0; i < spansBySentencePos.length; i++){
+        const span = spansBySentencePos[i];
+        if (index >= span.begin && index < span.end)
+            return false;
+    }
+    return true;
+
+}
+
+function findSubstring(sequence, substring, spansBySentencePos) {
     for (let i = 0; i < sequence.length - substring.length + 1; i++) {
         let found = true;
+
         for (let j = 0; j < substring.length; j++) {
             if (sequence[i+j] !== substring[j]) {
                 found = false;
                 break;
             }
         }
-        if (found)
+        if (found && check_range(i, spansBySentencePos))
```

Comment:
This (non-overlapping spans) and the sorting by span length are the only functional changes, correct? The rest is just code cleanup?

Reply:
yeah. I found the bug. It was not correctly reverting the position orders. it should be fixed now.